### PR TITLE
Make `--source-version` required

### PR DESCRIPTION
**Pull request by peternied**
### Description
By defaulting to ES 7.10 this caused confusion with the version parameter was not passed into RFS for different snapshot sources.  While this does require all customers to follow the same path it will prevent strange errors when the snapshot reader is unable to parse the snapshot.

### Issues Resolved
- Need to create

### Check List
- [ ] ~New functionality includes testing~
  - [ ] ~All tests pass, including unit test, integration test and doctest~
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: deployment/cdk/opensearch-service-migration/cdk.context.json
Comment (AndreKurait) [inline deployment/cdk/opensearch-service-migration/cdk.context.json:1]: @peternied, i was looking at the options.md to see if it needed to be updated, and it looks like `version` has already been implemented as part of the sourceCluster and targetCluster above on a previous PR you reviewed.

Can we modify this implementation to suggest specifying it there and update the options.md to say it's required on the source and target if using compatibility mode?

It would also be nice to throw an error during parsing the sourceCluster object if it's not specified.

https://github.com/opensearch-project/opensearch-migrations/pull/960

**Review by AndreKurait — CHANGES_REQUESTED** [review]

File: deployment/cdk/opensearch-service-migration/cdk.context.json
Comment (peternied) [reply deployment/cdk/opensearch-service-migration/cdk.context.json:1]: Thanks for looking into this, I've updated the options.md and examples to be correct following the format expected by the python system.

I did not throw an error on a missing value, it broke a sufficiently large number of test cases and I'd rather we reorganized how the validation of these parameters is handled that duplicate the logic in python typescript and java code in a future PR.

File: deployment/cdk/opensearch-service-migration/lib/common-utilities.ts [hunk 1]
```diff
@@ -417,10 +417,10 @@ function parseAuth(json: any): ClusterAuth | null {
 
 export function parseClusterDefinition(json: any): ClusterYaml {
     const endpoint = json.endpoint
-    const version = json.version
     if (!endpoint) {
         throw new Error('Missing required field in cluster definition: endpoint')
     }
+    const version = json.version;
```
Comment (AndreKurait) [inline deployment/cdk/opensearch-service-migration/lib/common-utilities.ts:423]: can we do a check in stack-composer that this version exists on the source object since it's required so it's a synth failure?

Comment (peternied) [reply deployment/cdk/opensearch-service-migration/lib/common-utilities.ts:423]: I did not throw an error on a missing value, it broke a sufficiently large number of test cases and I'd rather we reorganized how the validation of these parameters is handled that duplicate the logic in python typescript and java code in a future PR.

**Review by AndreKurait — APPROVED** [review]
